File: frontend-jobflow/src/app/core/services/company.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { environment } from '../../../environments/environment';
import { Company, CompanyRequest, Page } from '../models/company.model';

@Injectable({ providedIn: 'root' })
export class CompanyService {
  private readonly http = inject(HttpClient);
  private readonly baseUrl = `${environment.apiBaseUrl}/companies`;

  list(params: { search?: string; industry?: string; favorite?: boolean; page?: number; size?: number } = {}) {
    let httpParams = new HttpParams();
    Object.entries(params).forEach(([key, value]) => {
      if (value !== undefined && value !== null && value !== '') {
        httpParams = httpParams.set(key, String(value));
      }
    });
    return this.http.get<Page<Company>>(this.baseUrl, { params: httpParams });
  }

  get(id: string) {
    return this.http.get<Company>(`${this.baseUrl}/${id}`);
  }

  create(request: CompanyRequest) {
    return this.http.post<Company>(this.baseUrl, request);
  }

  update(id: string, request: CompanyRequest) {
    return this.http.put<Company>(`${this.baseUrl}/${id}`, request);
  }

  delete(id: string) {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  toggleFavorite(id: string) {
    return this.http.patch<Company>(`${this.baseUrl}/${id}/favorite`, {});
  }
}

File: backend-jobflow/src/main/java/com/jobflow/service/CompanyService.java
package com.jobflow.service;

import com.jobflow.dto.request.CompanyRequest;
import com.jobflow.dto.response.CompanyResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface CompanyService {
    CompanyResponse create(CompanyRequest request);
    CompanyResponse update(UUID id, CompanyRequest request);
    void delete(UUID id);
    CompanyResponse get(UUID id);
    Page<CompanyResponse> list(String search, String industry, Boolean favorite, Pageable pageable);
    CompanyResponse toggleFavorite(UUID id);
}
